src/gluonts/distribution/transformed_distribution.py

```diff
@@ -129,14 +129,17 @@ def quantile(self, level: Tensor) -> Tensor:
             sign = sign * t.sign
 
         if not isinstance(sign, (mx.nd.NDArray, mx.sym.Symbol)):
-            sign = sign + level.zeros_like()
-
-        cond = F.broadcast_greater(sign, sign.zeros_like())
-        level = F.broadcast_mul(cond, level) + F.broadcast_mul(
-            1.0 - cond, 1.0 - level
-        )
+            level = level if sign > 0 else (1.0 - level)
+            q = self.base_distribution.quantile(level)
+        else:
+            q_pos = self.base_distribution.quantile(level)
+            q_neg = self.base_distribution.quantile(1.0 - level)
+            cond = F.broadcast_greater(sign, sign.zeros_like())
+            cond = F.broadcast_add(
+                cond.expand_dims(axis=1), q_pos.zeros_like()
+            )
+            q = F.where(cond, q_pos, q_neg)
```

Comment:
This makes two calls to `.quantile` instead of one. Wouldn't something like

```python
cond = F.broadcast_greater(sign, sign.zeros_like())
signed_level = F.where(cond, level, 1.0 - level)
q = self.base_distribution.quantile(signed_level)
```

be more efficient?

Reply:
This is what we had before. The problem is that in the current API for `.quantile` `level` is supposed to be a 1d tensor of level values, i.e., it doesn't know about the shape of the distribution. The previous code blew up the `level` to a larger tensor mixing in the shape of `sign`. It did work (not entirely sure how), but it resulted in very large tensors during the `quantile` computation that don't fit in GPU memory for e.g. `Binned` distribution when the distribution dimension was large. Maybe we can rethink the api for `quantile` to not only support 1d level values, but also level values per distribution dimension.

Reply:
Makes sense, thanks for clarifying. If that’s the contract that `quantile` assumes (scalar or 1D tensor) then I agree that your solution is probably the best.

In a separate story, we may want to start shape assertions in order to avoid these weird situations where things accidentally work :-) (maybe because of some strange broadcasting coincidence)